lib/rules/no-underscore-dangle.js

```diff
@@ -162,14 +166,57 @@ module.exports = {
             }
         }
 
+        /**
+         * Check if method declaration has a underscore at the end
+         * @param {ASTNode} node node to evaluate
+         * @returns {void}
+         * @private
+         */
+        function checkForTrailingUnderscoreInMethodDefinition(node) {
+            const identifier = node.key.name;
+
+            if (typeof identifier !== "undefined" && hasTrailingUnderscore(identifier) && enforceInMethodNames) {
```

Comment:
I feel like the `enforceInMethodNames` guard should come first in this conditional so that `hasTrailingUnderscore()` isn't called every time.